### PR TITLE
My Site Dashboard: Update blog for site picker

### LINKED ISSUE #17982: Title and URL remain after deleting WP.com site
This was tested with a WP.com account that had access to multiple sites.

### Expected behavior
After site deletion, the app switches to an exiting site and updates the title and URL to the site it switched to.

### Actual behavior
After site deletion, the app switches to an exiting site but doesn't update the title and URL to the site it switched to.

### Steps to reproduce the behavior
1. On a WP.com site that's OK to delete, go to Site Settings from the My Site view
2. Delete Site
3. Observe that the title and URL of the deleted site remain in the My Site view

https://user-images.githubusercontent.com/2092798/154616284-94c8c71f-b6cf-4456-ae6b-de6fb47b7467.mp4

### LINKED ISSUE #17993: Title and URL aren't updated after creating WP.com site
### Expected behavior
After creating a new WP.com site, the site title and URL are updated on the My Site view.

### Actual behavior
After creating a new WP.com site, the site title and URL aren't updated on the My Site view.

### Steps to reproduce the behavior
1. Have an existing site loaded in the My Site view
2. Create a new WP.com site
3. Observe that the title and URL of the previous site remain in the My Site view

https://user-images.githubusercontent.com/2092798/154783951-5be03345-41df-40d5-a738-8b2f7b446977.mp4

### PR DESCRIPTION
**Pull request by momo-ozawa**
Fixes #17993 
Fixes #17982

## Description

- Fixes an issue where the blog associated with the SitePickerViewController wasn't being updated to the correct blog for site creation / site deletion

### Site creation
1. Have an existing site loaded in the My Site view
2. Create a new WP.com site
3. ✅ Observe that the title and URL reflect the newly created site

https://user-images.githubusercontent.com/6711616/154847950-8dc93260-aa31-4773-ae18-d3d891b2214b.mp4

### Site deletion (account w multiple sites)
1. On a WP.com site that's OK to delete, go to Site Settings from the My Site view
2. Delete Site
3. ✅ Observe that the title and URL have changed to a different, existing site

https://user-images.githubusercontent.com/6711616/154847961-49ca9749-9d91-4149-acec-7136a5d63a4a.mp4

### Site deletion (account w one site)
1. On a WP.com site that's OK to delete, go to Site Settings from the My Site view
2. Delete Site
3. ✅ Observe that "Create a new site for your business, magazine, or personal blog; or connect an existing WordPress installation" is displayed

https://user-images.githubusercontent.com/6711616/154847983-e3c58886-369a-4f2e-b031-ae512674c5b7.mp4

## Regression Notes
1. Potential unintended areas of impact
n/a

2. What I did to test those areas of impact (or what existing automated tests I relied on)
n/a

3. What automated tests I added (or what prevented me from doing so)
n/a

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by momo-ozawa** [comment]
@mokagio @leandroalonso 
This is a fix for a [regression introduced in 19.2](p5T066-2YK-p2#comment-11372) that @twstokes found yesterday.

While 19.2 is due to be released on Monday Feb 21, would it be possible to bundle these changes as part of 19.2? Since the 19.2 milestone is closed, I’ve selected Pending as the milestone for now.

**Review by twstokes — APPROVED** [review]
The code LGTM and I tested both (which are the first two test cases):
- https://github.com/wordpress-mobile/WordPress-iOS/issues/17993
- https://github.com/wordpress-mobile/WordPress-iOS/issues/17982

And both were resolved. 🎉 I also tested the third case (site deletion with one site) and it passed as well.

Thanks so much @momo-ozawa for jumping on this quickly!

**Comment by mokagio** [comment]
I changed the base for this to `release/19.2.1` because I've decided / had to make an hotfix for this. More details in the internal discussion at https://wp.me/p5T066-2YK#comment-11376.

TL;DR:

- I needed 19.2 into `trunk` to move on with the scheduled 19.3 code freeze
- I also needed 19.2 released (via GitHub release) to start the 19.2.1 hotfix
- It more straightforward to followup with an hotfix rather than scrapping the finalized 19.2 artifacts and "re-finalize"
- This makes sense to me as an approach because the issue was not resulting in a crash, we can still phase release 19.2 if we want without completely breaking the users' experience.

**Comment by mokagio** [comment]
Taking over the merge after @twstokes approval to move forward with the hotfix procedure. Also notice the base for this PR is now `release/19.2.1`
